### DISCUSSION TITLE
Using the new Allure-Cypress Plugin

### DISCUSSION THREAD

**Discussion by achraf-elk** [badeball/cypress-cucumber-preprocessor#1191] [Q&A]
I've been encountering an issue when running Cypress tests with the cypress-cucumber-preprocessor alongside the new [allure-cypress](https://allurereport.org/docs/cypress/) plugin. The error occurs during the before:spec event handler and causes my test suite to fail after running some tests. Here is the error message:

```
An error was thrown in your plugins file while executing the handler for the before:spec event.
The error we received was:
Error: Unexpected state in beforeSpecHandler: test-finished. This almost always means that you or some other plugin, are overwriting this plugin's event handlers. For more information & workarounds, see https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/event-handlers.md (if neither workaround work, please report at https://github.com/badeball/cypress-cucumber-preprocessor)
```

I understand from this [discussion ](https://github.com/orgs/allure-framework/discussions/2409 )that the current allure-cypress plugin doesn't not support Cucumber yet. Could you provide any insights on whether this issue might be resolved with future updates to either the allure-cypress or cypress-cucumber-preprocessor plugins?

Is there any recommended workaround for this issue in the meantime?

**Comment by badeball** [comment]
This is explained in the [documentation ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/event-handlers.md) linked from the error message. Use `cypress-on-fix`.

> Could you provide any insights on whether this issue might be resolved with future updates to either the allure-cypress or cypress-cucumber-preprocessor plugins?

No, I don't have any insight into what allure is nor what their plugin is or isn't doing.
